core/Cargo.toml

```diff
@@ -19,18 +19,28 @@ doctest = false
 
 [features]
 default = ["std"]
+diagnostics = ["dep:miette"]
 std = [
-	"miden-crypto/std",
-	"miden-formatting/std",
-	"math/std",
-	"winter-utils/std",
-	"thiserror/std",
+    "dep:parking_lot",
+    "memchr/std",
+    "miden-crypto/std",
+    "miden-formatting/std",
+    "math/std",
+    "winter-utils/std",
+    "thiserror/std",
 ]
 
 [dependencies]
 math = { package = "winter-math", version = "0.9", default-features = false }
+memchr = { version = "2.7.4", default-features = false }
```

Comment:
nit here and for other new dependencies: we usually omit patch part of versions.